modules/core/src/shaderlib/project/project.glsl.ts

```diff
@@ -36,23 +36,28 @@ ${COORDINATE_SYSTEM_GLSL_CONSTANTS}
 ${PROJECTION_MODE_GLSL_CONSTANTS}
 ${UNIT_GLSL_CONSTANTS}
 
-uniform int project_uCoordinateSystem;
-uniform int project_uProjectionMode;
-uniform float project_uScale;
-uniform bool project_uWrapLongitude;
-uniform vec3 project_uCommonUnitsPerMeter;
-uniform vec3 project_uCommonUnitsPerWorldUnit;
-uniform vec3 project_uCommonUnitsPerWorldUnit2;
-uniform vec4 project_uCenter;
-uniform mat4 project_uModelMatrix;
-uniform mat4 project_uViewProjectionMatrix;
-uniform vec2 project_uViewportSize;
-uniform float project_uDevicePixelRatio;
-uniform float project_uFocalDistance;
-uniform vec3 project_uCameraPosition;
-uniform vec3 project_uCoordinateOrigin;
-uniform vec3 project_uCommonOrigin;
-uniform bool project_uPseudoMeters;
+uniform project32Uniforms {
```

Comment:
Nit: Can we keep module and UBO name consistent? Do we have a good reason to diverge?
```suggestion
uniform projectUniforms {
```

Reply:
This is what I meant by "hacky"...

The `project32` module depends on `project`. I would like to use `uniform projectUniforms {...` but currently that won't work in luma as the `ShaderInputs` doesn't traverse the module depencies. So as I first step I tried to port the shaders with workarounds [like this](https://github.com/visgl/deck.gl/pull/8782/files#diff-9dc0b8596c862e17e7ef398d73fd9f6e0c1ea1865c0d8e26ead182080c2b3956R51-R53), but the view to then see what is required in luma to land this cleanly.

If you'd like to look into how to solve this from the luma side, that would be appreciated. We will have this issue with other modules also

Reply:
Implemented in luma: https://github.com/visgl/luma.gl/pull/2073